Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -196,27 +200,53 @@ public struct BSONBinary: BSONValue, Equatable, Codable, Hashable {
     internal var bson: BSON { .binary(self) }
 
     /// The binary data.
-    public let data: Data
+    public let data: ByteBuffer
 
     /// The binary subtype for this data.
-    public let subtype: UInt8
+    public let subtype: Subtype
 
     /// Subtypes for BSON Binary values.
-    public enum Subtype: UInt8 {
+    public struct Subtype: Equatable, Codable, Hashable, RawRepresentable {
         /// Generic binary subtype
-        case generic,
-            /// A function
-            function,
-            /// Binary (old)
-            binaryDeprecated,
-            /// UUID (old)
-            uuidDeprecated,
-            /// UUID (RFC 4122)
-            uuid,
-            /// MD5
-            md5,
-            /// User defined
-            userDefined = 0x80
+        public static let generic = Subtype(0x00)
+        /// A function
+        public static let function = Subtype(0x01)
+        /// Binary (old)
+        public static let binaryDeprecated = Subtype(0x02)
+        /// UUID (old)
+        public static let uuidDeprecated = Subtype(0x03)
+        /// UUID (RFC 4122)
+        public static let uuid = Subtype(0x04)
+        /// MD5
+        public static let md5 = Subtype(0x05)
+        /// Encrypted BSON value
+        public static let encryptedValue = Subtype(0x06)
+
+        /// Subtype indicator value
+        public let rawValue: UInt8
+
+        /// Initializes a `Subtype` with a custom value.
+        /// Returns nil if rawValue outside of the range 0x80-0xFF.
+        public init?(rawValue: UInt8) {
+            guard rawValue >= 0x80 && rawValue <= 0xFF else {
```

Comment:
This validation should only omit the values that are reserved but not allocated (e.g. `[0x07, 0x80)`)